plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -50,6 +171,155 @@ public void testVarbinary()
                 .addRoundTrip("varbinary", "X'0001020304050607080DF9367AA7000000'", VARBINARY, "X'0001020304050607080DF9367AA7000000'") // non-text
                 .addRoundTrip("varbinary", "X'000000000000'", VARBINARY, "X'000000000000'")
                 .execute(getQueryRunner(), trinoCreateAsSelect("test_varbinary"));
+
+        // Binary literals must be prefixed with 0x
+        // https://docs.microsoft.com/en-us/sql/analytics-platform-system/load-with-insert?view=aps-pdw-2016-au7#InsertingLiteralsBinary
+        SqlDataTypeTest.create()
+                .addRoundTrip("varbinary(10)", "NULL", VARBINARY, "CAST(NULL AS varbinary)")
+                .addRoundTrip("varbinary(20)", "0x", VARBINARY, "CAST('' AS varbinary)")
+                .addRoundTrip("varbinary(30)", "0x68656C6C6F", VARBINARY, "CAST(X'68656C6C6F' AS varbinary)")
+                .addRoundTrip("varbinary(1000)", "0x5069C4996B6E6120C582C4856B61207720E69DB1E4BAACE983BD", VARBINARY, "CAST(X'5069C4996B6E6120C582C4856B61207720E69DB1E4BAACE983BD' AS varbinary)")
+                .addRoundTrip("varbinary(2000)", "0x4261672066756C6C206F6620F09F92B0", VARBINARY, "CAST(X'4261672066756C6C206F6620F09F92B0' AS varbinary)")
+                .addRoundTrip("varbinary(4000)", "0x0001020304050607080DF9367AA7000000", VARBINARY, "CAST(X'0001020304050607080DF9367AA7000000' AS varbinary)") // non-text
```

Comment:
The `CAST(... as varbinary)` is redundant, since `X'...'` literal is already `varbinary`.